internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,145 @@
+package olm
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	core "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+// RegistryPod holds resources necessary for creation of a registry server
+type RegistryPod struct {
+	registryImage       string
+	registryPodGRPCPort int32
+	registryDBPath      string
+	registryBundleImage string
+
+	ctx        context.Context
+	kubeclient kubernetes.Interface
+}
+
+// GetKubeClient creates a kubernetes clientset using kubeconfig
+func GetKubeClient() (client kubernetes.Interface) {
+	var kubeconfig *string
+	if home := homedir.HomeDir(); home != "" { // Check if machine has home directory
+		// Read kubeconfig flag. If not provided, use config file $HOME/.kube/config
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	// Build configuration from the config file
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create kubernetes clientset, which can be used to create, delete, patch, list kubernetes resources
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+
+	return clientset
+}
+
+// CreateBundleRegistry returns a registry pod built from an index image
+func (rp *RegistryPod) CreateBundleRegistry() *core.Pod {
+	var pod *core.Pod
+
+	// TODO: Customizable fields can be moved to RegistryPod
+	podObject := &core.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "bundle-registry-pod",
+			Namespace: "default",
+			Labels: map[string]string{
+				"app": "default",
+			},
+		},
+		Spec: core.PodSpec{
+			Containers: []core.Container{
+				{
+					Name:    "registry-grpc",
+					Image:   rp.registryImage,
+					Command: []string{"/bin/sh"},
+					Args: []string{
+						"-c",
+						getContainerCmd(rp.registryDBPath, rp.registryBundleImage),
+					},
+					Ports: []core.ContainerPort{
+						{Name: "registry-port", ContainerPort: rp.registryPodGRPCPort},
+					},
+				},
+			},
+		},
+	}
+
+	// Check if Pod already exists
+	pod, err := rp.kubeclient.CoreV1().Pods(podObject.Namespace).Get(rp.ctx, podObject.Name, metav1.GetOptions{})
+	if err != nil && errors.IsNotFound(err) {
```

Comment:
Need to generally check if `err != nil` then check `errors.IsNotFound(err)` in case the error is for some other reason.

Reply:
You're saying we need to handle the case where `err != nil` and it is something other than `errors.IsNotFound()`, right? If so, I agree :)

Basically:
```go
if err != nil {
    if errors.IsNotFound(err) {
         // create it
    }
    // otherwise, return the error
    return nil, err
}
```